autokey-blueprints/1-storage-bucket/main.tf

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  default_suffix              = var.suffix != "" ? var.suffix : random_string.suffix.result
+  autokey_resource_project_id = var.create_autokey_resource_project ? module.autokey_resource_project[0].project_id : var.autokey_resource_project_id
+}
+
+resource "random_string" "suffix" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+module "autokey" {
```

Comment:
Good point. Probably terraform is noticing the implicit dependency since we are consuming the project name from `local.autokey_resource_project_id`, but let's make explicit to avoid any trouble.